llvm/lib/Transforms/Scalar/LoopIdiomRecognize.cpp

```diff
@@ -411,14 +421,12 @@ static Constant *getMemSetPatternValue(Value *V, const DataLayout *DL) {
   if (Size > 16)
     return nullptr;
 
-  // If the constant is exactly 16 bytes, just use it.
-  if (Size == 16)
-    return C;
+  // For now, don't handle types that aren't int, floats, or pointers.
+  if (!isa<ConstantInt>(C) && !isa<ConstantFP>(C) &&
```

Comment:
Better check the type for these, otherwise it's going to implicitly allow vector splats in the future...